presto-spi/src/main/java/com/facebook/presto/spi/statistics/ComputedStatistics.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import com.facebook.presto.spi.block.Block;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+
+public class ComputedStatistics
+{
+    private final List<String> groupingColumns;
+    private final List<Block> groupingValues;
+    private final Map<TableStatisticType, Block> tableStatistics;
+    private final Map<ColumnStatisticMetadata, Block> columnStatistics;
+
+    private ComputedStatistics(
+            List<String> groupingColumns,
+            List<Block> groupingValues,
+            Map<TableStatisticType, Block> tableStatistics,
+            Map<ColumnStatisticMetadata, Block> columnStatistics)
+    {
+        this.groupingColumns = unmodifiableList(new ArrayList<>(requireNonNull(groupingColumns, "groupingColumns is null")));
```

Comment:
Is there any reason not using `ImmutableList.copyOf()` ?

Reply:
`ImmutableList` is a `Guava` class. We don't have Guava in the classpath of the `presto-spi` module.